### ISSUE TITLE
/etc/zookeeper not created

### ISSUE THREAD

**Issue by skpy** [deric/puppet-zookeeper#96]
Trying the archive installation, the Puppet run failed:
```
Info: Applying configuration version '00066f687102c86be6f1d20953babda4c3da6b15'
Notice: /Stage[main]/Zookeeper::Install/Package[java-1.7.0-openjdk-headless]/ensure: created
Notice: /Stage[main]/Zookeeper::Install::Archive/Archive[/opt/zookeeper-3.4.10.tar.gz]/ensure: download archive from http://apache.org/dist/zookeeper/zookeeper-3.4.10/zookeeper-3.4.10.tar.gz to /opt/zookeeper-3.4.10.tar.gz and extracted in /opt with cleanup
Info: /Stage[main]/Zookeeper::Install::Archive/Archive[/opt/zookeeper-3.4.10.tar.gz]: Scheduling refresh of Exec[chown_zookeeper_directory]
Notice: /Stage[main]/Zookeeper::Install::Archive/Exec[chown_zookeeper_directory]: Triggered 'refresh' from 1 events
Notice: /Stage[main]/Zookeeper::Install::Archive/File[/opt/zookeeper]/ensure: created
Error: Cannot create /etc/zookeeper/conf; parent directory /etc/zookeeper does not exist
Error: /Stage[main]/Zookeeper::Config/File[/etc/zookeeper/conf]/ensure: change from absent to directory failed: Cannot create /etc/zookeeper/conf; parent directory /etc/zookeeper does not exist
```
Should this module create the `/etc/zookeeper` directory before trying to create the `conf` sub-directory?

**Comment by deric** [comment]
Yes, it should. The config directory is created with [`recurse => true`](https://github.com/deric/puppet-zookeeper/blob/master/manifests/config.pp#L13). In normal case this should ensure existence of `/etc/zookeeper` when `$cfg_dir` is set to `/etc/zookeeper/conf`.

Which version of `deric-zookeeper` do you use? And which version of Puppet? Is `/etc/` writeable for user executing Puppet?

**Comment by skpy** [comment]
I don't think [recurse](https://docs.puppet.com/puppet/4.3/types/file.html#file-attribute-recurse) works that way.  The documentation states that this directive controls
> Whether to recursively manage the *contents* of a directory.

What you're describing is basically `mkdir -p`, which Puppet doesn't support very well. At least, it's never worked well for me.

Puppet is running at root, and /etc is writable by root.  We're using Puppet (Enterprise) 4.5.3, and 0.7.2 of this module from the Forge.

I've worked around this problem by defining a File resource for `/etc/zookeeper` in the relevant profile.  Thanks for this module - it works great otherwise!

**Comment by deric** [comment]
Yes, you're right. With "standard" binary installation `/etc/zookeeper` is created when ZooKeeper package is installed, so we don't have to care about that. We could check that the parent directory of `$cfg_dir` exists, but that could lead to duplicate declarations in case that someone changes `$cfg_dir`.

**Comment by deric** [comment]
Should be fixed in [v0.7.3](https://github.com/deric/puppet-zookeeper/releases/tag/v0.7.3).
